GillianCore/engine/Abstraction/Verifier.ml

```diff
@@ -453,6 +484,40 @@ struct
         in
         Printf.printf "%s\n" msg;
         L.normal (fun m -> m "%s" msg)
+
+  let prune_call_graph call_graph proc_names =
+    let proc_ids = List.map CallGraph.id_of_proc_name proc_names in
+    List.iter (CallGraph.remove call_graph) proc_ids
+
+  let prune_results results proc_names =
+    List.iter (VerificationResults.remove results) proc_names
+
+  let verify_prog (prog : (Annot.t, int) Prog.t) (incremental : bool) : unit =
```

Comment:
I'll take a much closer look at this, you'll walk me through it step by step on zoom 😅